lm_eval/arguments.py

```diff
@@ -41,3 +41,9 @@ class EvalArguments:
     seed: Optional[int] = field(
         default=0, metadata={"help": "Random seed used for evaluation."}
     )
+    num_return_sequences: Optional[int] = field(
+        default=1,
+        metadata={
+            "help":"The number of independently computed return sequences for each element in the batch"
+            }
+    )
```

Comment:
Why do we need this argument + `n_samples` - Aren't they kind of the same?

Reply:
The `n_samples` argument is to capture the overall number of samples to be generated for a prompt/task. While the `num_return_sequences` is for the number of samples to be generated in one single pass. 

There can be scenarios when `n_samples` > `num_return_sequences` , like when the `n_samples` does not fit in the memory. In that case, the task/prompt is repeated ( multiple passes) to meet the overall `n_samples` ( as implemented [here](https://github.com/bigcode-project/bigcode-evaluation-harness/blob/8f9fae1e794b7b02cceb1afd13d1bf42cc13c9ca/lm_eval/generation.py#L71))

For example, to calculate pass@100 I might need the `n_samples` to be 100 and due to memory limit I can have `num_return_sequences` to be 10, so the task is repeated 10 times to meet the `n_samples` count of 100.

Reply:
But shouldn't the batch_size be responsible for handling the memory limitations? Can't we use it to infer  `num_return_sequences`?

IIURC it means that n_samples=16 batch_size=16 num_return_sequences=1 is the same as n_samples=16 batch_size=1 num_return_sequences=16, right?

Reply:
I agree that both the settings are same for the case that you have shown. But I am not quite getting how we can infer `num_return_sequences` from batch_size. Can you please explain ? Thanks

Reply:
IIURC batch_size is used to pick batch_size new items, so I think sth like:
```python
if batch_size < n_samples:
    # Memory requirement will be the same as batch_size, but we only pick 1 new item (i.e. batch_size=1)
    num_return_sequences = batch_size
    batch_size = 1
else:
    # If n_samples, just pick batch_size new items; If n_samples > 1, somewhere in-between
    num_return_sequences = n_samples
    # Round down, such that we always have <= batch_size items in one go
    batch_size = batch_size // num_return_sequences
```